src/test/javascript/integration/jvm_metrics_service_spec.js

```diff
@@ -0,0 +1,119 @@
+const bodyParser = require('body-parser');
+const express = require('express');
+const http = require('http');
+const {
+    closeServer,
+    fromObjectPath,
+    getProp,
+    hubClientGet,
+    itSleeps,
+} = require('../lib/helpers');
+const { getHubUrlBase, getIp } = require('../lib/config');
+
+const headers = {
+    'Content-Type': 'application/json',
+    Accept: "application/json",
+};
+const sharedContext = {
+    queryResults: [],
+    payloadResults: [],
+    server: null,
+};
+
+const startMockInfluxDb = async () => {
+    const app = express();
+
+    app.use(bodyParser.raw({type: 'application/x-www-form-urlencoded', limit: '2mb' }));
+
+    app.post('/write', (request, response) => {
+        const body = getProp('body',request);
+        const metrics = Buffer.from(body).toString();
+        const query = fromObjectPath(['query', 'db'], request);
+        const metricsArray = metrics && metrics.split('\n');
+        sharedContext.queryResults.push(query);
+        sharedContext.payloadResults.push(...metricsArray);
+        response.sendStatus(200);
+    });
+
+    const server =  new http.Server(app);
+
+    server.on('connection', (socket) => {
+        socket.setTimeout(1000);
+    });
+
+    server.on('request', function (request, response) {
+       request.on('end', function () {
+            response.end();
+        });
+    });
+
+    const callbackDomain = getIp();
+
+    server.listen(8086, callbackDomain);
+    sharedContext.server = await server.on('listening', () => {
+        console.log(`server listening at http://${getIp()}:8086/write`);
+        return server;
+    });
+};
+
+const EXPECTED_METRIC_TAGS = [
+    'jvm_gc.G1-Old-Generation',
+    'jvm_gc.G1-Young-Generation',
+    'jvm_memory.heap',
+    'jvm_memory.non-heap',
+    'jvm_memory.pools',
+    'jvm_memory.total',
+    'jvm_thread',
+];
+
+const EXPECTED_HOST_TAGS = [
+    'cluster=local-single',
+    'env=single',
+    'role=hub',
+    'team=ddt',
+    'version=local',
+];
+
+describe(__filename, function () {
+    beforeAll(async () => {
+        pending('infrastructure needed to allow metrics.enable to be dynamically true in test envs');
```

Comment:
NOTE: this is requires some separate infra changes that I think are out of the scope of this PR, and may be better as one of the tests that is run against an ephemeral cluster constructed for the single test. I see value in the test being written and passing locally in the mean time. Leaving it pending for now...